plugins/filter/filters.py

```diff
@@ -32,46 +33,67 @@ def filters(self):
             'resolve_principal': self.resolve_principal
         }
 
-    def normalize_sasl_protocol(self, protocol):
-        # Returns standardized value for sasl mechanism string
-        normalized = 'GSSAPI' if protocol.lower() == 'kerberos' \
-            else 'SCRAM-SHA-512' if protocol.upper() == 'SCRAM' \
-            else 'SCRAM-SHA-256' if protocol.lower() == 'scram256' \
-            else 'PLAIN' if protocol.upper() == 'PLAIN' \
-            else 'OAUTHBEARER' if protocol.upper() == 'OAUTH' \
-            else 'none'
-        return normalized
-
-    def kafka_protocol_normalized(self, sasl_protocol_normalized, ssl_enabled):
+    def normalize_sasl_protocol(self, protocols):
+        # Returns a list of standardized values for sasl mechanism strings
+        protocol_list = protocols.replace(" ", "").split(',')
+
+        normalized_protocols = []
+        for protocol in protocol_list:
+            if protocol.lower() == 'kerberos':
+                normalized = 'GSSAPI'
+            elif protocol.upper() == 'SCRAM':
+                normalized = 'SCRAM-SHA-512'
+            elif protocol.lower() == 'scram256':
+                normalized = 'SCRAM-SHA-256'
+            elif protocol.upper() == 'PLAIN':
+                normalized = 'PLAIN'
+            elif protocol.upper() == 'OAUTH':
+                normalized = 'OAUTHBEARER'
+            else:
+                normalized = 'none'
+            normalized_protocols.append(normalized)
+        return normalized_protocols
+
+    def kafka_protocol_normalized(self, sasl_protocols_normalized, ssl_enabled):
         # Joins a sasl mechanism and tls setting to return a kafka protocol
-        kafka_protocol = 'SASL_SSL' if ssl_enabled is True \
-            and sasl_protocol_normalized in ['GSSAPI', 'PLAIN', 'SCRAM-SHA-512', 'SCRAM-SHA-256', 'OAUTHBEARER'] \
-            else 'SASL_PLAINTEXT' if ssl_enabled is False \
-            and sasl_protocol_normalized in ['GSSAPI', 'PLAIN', 'SCRAM-SHA-512', 'SCRAM-SHA-256', 'OAUTHBEARER'] \
-            else 'SSL' if ssl_enabled is True and sasl_protocol_normalized == 'none' \
-            else 'PLAINTEXT'
+        required_mechanisms = ['GSSAPI', 'PLAIN', 'SCRAM-SHA-512', 'SCRAM-SHA-256', 'OAUTHBEARER']
+
+        if ssl_enabled and self.all_elements_present(sasl_protocols_normalized, required_mechanisms):
```

Comment:
How do we want to handle the case where some element is in required mechanisms and some isnt?

Reply:
Only combinations which are allowed when having multiple values is [plain,kerberos,scram,scram256]
Validation for the same has been added.